nifi-nar-bundles/nifi-asana-bundle/nifi-asana-services/src/main/java/org/apache/nifi/controller/asana/StandardAsanaClient.java

```diff
@@ -0,0 +1,303 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.controller.asana;
+
+import com.asana.Client;
+import com.asana.errors.InvalidTokenError;
+import com.asana.models.Attachment;
+import com.asana.models.Event;
+import com.asana.models.Project;
+import com.asana.models.ProjectMembership;
+import com.asana.models.ProjectStatus;
+import com.asana.models.Resource;
+import com.asana.models.ResultBodyCollection;
+import com.asana.models.Section;
+import com.asana.models.Story;
+import com.asana.models.Tag;
+import com.asana.models.Task;
+import com.asana.models.Team;
+import com.asana.models.User;
+import com.asana.models.Workspace;
+import com.asana.requests.CollectionRequest;
+import com.asana.requests.EventsRequest;
+import com.google.gson.annotations.SerializedName;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class StandardAsanaClient implements AsanaClient {
+
+    static final String ASANA_CLIENT_OPTION_BASE_URL = "base_url";
+
+    private final Client client;
+    private final Workspace workspace;
+
+    public StandardAsanaClient(String personalAccessToken, String workspaceName, String baseUrl) {
+        client = Client.accessToken(personalAccessToken);
+        if (baseUrl != null) {
+            client.options.put(ASANA_CLIENT_OPTION_BASE_URL, baseUrl);
+        }
+        workspace = getWorkspaceByName(workspaceName);
+    }
+
+    @Override
+    public Project getProjectByName(String projectName) {
+        return getProjects()
+                .values()
+                .stream()
+                .filter(p -> p.name.equals(projectName))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("No such project: " + projectName));
+    }
+
+    @Override
+    public Map<String, Project> getProjects() {
+        try {
+            return collectionRequestToMap(
+                    client.projects.getProjects(null, null, workspace.gid, null, null, getSerializedFieldNames(Project.class), false)
+            );
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
`RuntimeException` should be replaced with `UncheckedIOException` to provide a better representation of the wrapped exception.

Reply:
Done in [c33ff0f](https://github.com/apache/nifi/pull/6504/commits/c33ff0fd9ab9cee55d4c921ebd54740883235618)